posts/2021-11-26-jakarta-ee-9.1.adoc

```diff
@@ -0,0 +1,433 @@
+
+Any Liberty features with API and/or SPI functions that use Jakarta EE APIs as part of their method signatures have been updated to have their package versions be reset to 10.0 when using those API / SPIs with Jakarta EE 9.1.  Any bundles used for user features that depend on those packages will need to change the import package version range when updating the user feature to use Jakarta EE 9.1.
+
+With the introduction of Jakarta EE 9.1, the Jakarta Enterprise Beans 4.0 specification includes a few minor changes over the prior version of the specification, Enterprise JavaBeans (EJB) 3.2, as follows:
+  
+*  Note the new names of the features; all of the same features exist, but the feature name prefix has changed from `ejb` to `enterpriseBeans`.  For example, `enterpriseBeansLite-4.0` is the new version of `ejbLite-3.2`.
+* The API package has changed from `javax.ejb` to `jakarta.ejb`
+* The `@Schedule` annotation is now repeatable
+* The following API methods have been removed:
+** `javax.ejb.EJBContext.getCallerIdentity()` -> use `getCallerPrincipal()`
+** `javax.ejb.EJBContext.getEnvironment()` -> use `JNDI` lookup in `java:comp/env`
+** `javax.ejb.EJBContext.isCallerInRole(java.security.Identity)` -> use `isCallerInRole(String)`
+** `javax.ejb.SessionContext.getMessageContext()` (removed with `JAX-RPC`)
+   
+All other capabilities of Enterprise Beans remain the same as the prior specification version (3.2).
+
+Enable the Jakarta EE 9.1 features in your app’s `server.xml`. You can enable the individual features you want or you can just add the Jakarta EE 9.1 convenience feature to enable all of the Jakarta EE 9.1 features at once:
+
+[source,xml]
+----
+<featureManager>
+    <feature>jakartaee-9.1</feature>
+</featureManager>
+----
+
+Or you can add the Web Profile convenience feature to enable all of the Jakarta EE 9.1 Web Profile features at once:
+
+[source,xml]
+----
+<featureManager>
+    <feature>webProfile-9.1</feature>
+</featureManager>
+----
+
+Although many of the Jakarta EE 9.1 features have only received a version update, the majority have also had their name changed.  The following table lists the features for which both the short name and the version number are changed. To update one of these features for Jakarta EE 9.1, you must change both the feature short name and version number in your `server.xml` file.
+
+.Table 1: Jakarta EE 9.1 feature updates, short name and version
+|===
+|Jakarta EE 9.1 feature name | Java EE/Jakarta EE 8 short name and version | Jakarta EE 9.1 short name and version
+
+| Jakarta Enterprise Beans
+| feature:ejb-3.2[]
```

Comment:
What's with the feature:name[]?  It doesn't format at all when I am looking at the preview.  Seems wrong.   Am I missing something?

Reply:
rushed copy and paste; fixing now